gensim/models/deprecated/doc2vec.py

```diff
@@ -992,7 +992,7 @@ def __iter__(self):
             fname = os.path.join(self.dirname, fname)
             if not os.path.isfile(fname):
                 continue
-            for item_no, line in enumerate(utils.smart_open(fname)):
+            for item_no, line in enumerate(utils.open(fname, 'rb')):
```

Comment:
Can we put this inside a context manager? We're opening a file pointer and leaving it dangling here, it's poor practice.

Reply:
done